src/client.ts

```diff
@@ -347,26 +349,30 @@ class OptimizelyReactSDKClient implements ReactSDKClient {
     return await this.userContext.fetchQualifiedSegments(options);
   }
 
-  public setUser(userInfo: UserInfo): void {
-    // TODO add check for valid user
-    if (userInfo.id) {
-      this.user.id = userInfo.id;
-      this.isUserReady = true;
+  public async setUser(userInfo: UserInfo): Promise<void> {
+    this.isUserPromiseResolved = false;
+    this.isUserReady = true;
 
-      if (this._client) {
-        this.userContext = this._client.createUserContext(userInfo.id, userInfo.attributes);
-      } else {
-        logger.warn(
-          'Unable to create user context for user id "%s" because Optimizely client failed to initialize.',
-          this.user.id
-        );
-      }
+    //reset user info
+    this.user = default_user;
+
+    this.user.id = userInfo.id;
+
+    if (this._client) {
+      this.userContext = this._client.createUserContext(userInfo.id ?? undefined, userInfo.attributes);
```

Comment:
I do not see new test cases covering null userId cases.

Reply:
the this._client.createUserContext only accepts string or undefined. and this undefined is then replaced with viud in javascript

Reply:
hence whatever falsy value the client may provide the react-sdk will be passing them as undefined